_posts/2019-06-06-elixir-conf-eu-2019-review.md

```diff
@@ -0,0 +1,192 @@
+
+Regarding alarms, in case for example the previous issue occurs where we can’t control to the database, the system should provide feedback to the operators, by raising an alarm. This can work by having a watchdog process which constantly monitors the database status and toggle an alarm based on it. 
+
+Lastly, he talked about how our system should handle failure with external services by using circuit breakers (like [Fuse, a circuit breaker for erlang](https://github.com/jlouis/fuse)). This can depend on a per application basis, but we could always have a write-through cache, which means that you can write the last good external service response to an ETS table, which in case the downstream service is down, we can provide to the end user. 
+
+He closed his talk by saying that, by solving all those potential issues before hand, we can prevent a ton of issues happening in the future, that we have really powerful tools in our runtime, and we should take advantage of them to build more robust systems.
+
+[![](https://img.youtube.com/vi/lg7M0h9eoug/0.jpg)](https://www.youtube.com/watch?v=lg7M0h9eoug)
+
+### Lessons From our first trillion messages with Flow by John Mertens
+
+John on his talk give a real life example on how they use Flow in Change.org to process a ton of messages stored in SQS. They have a Ruby/JS website where people’s actions produce and store messages in SQS, which are then processed by their Elixir / Flow system. 
+
+The 3 lessons they have learned, are the following:
+
+1) Let Flow do the work
+2) Organize your Flow
+3) Tune your Flow
+
+In Lesson 1, we learn about GenStage / Flow and how their amazing features can help with a big long running system like that. Parallelism is made easy, and by using Flow, you are protected from DDoSing yourself, because instead of pushing from one system to another via stream, now you pull as many messages as you can get.
+
+In Lesson 2, John talked about how one should know their system pretty well, so that they can provide the correct configuration to it. For example, by using SQS, you are limited to receiving batches of 10 messages each time, so you have to build around that, unless you want to wait for say 500 messages to arrive at your system, 10 at a time. Another useful tip is to use a token, meaning a struct that gets pushed down through the whole system, changed at each step, which makes processing and pattern matching on messages really easy (it’s the same thing that Plug uses with it’s Conn struct). Errors are supposed to be passed through the whole flow, and left out of acknowledgement, so SQS will resend them, and you can process them again.
```

Comment:
"...unless you want to wait for say 500 messages to arrive at your system, 10 at a time."
Maybe it's me but this doesn't make sense to me. Can you explain please?